soh/soh/Enhancements/randomizer/randomizer_check_objects.cpp

```diff
@@ -0,0 +1,815 @@
+    { RC_LH_LAB_GOSSIP_STONE, {RC_LH_LAB_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_LAKE_HYLIA, "Lab Gossip Stone", "LH Lab Gossip Stone"}},
+    { RC_LH_SOUTHEAST_GOSSIP_STONE, {RC_LH_SOUTHEAST_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_LAKE_HYLIA, "Southeast Gossip Stone", "LH Southeast Gossip Stone"}},
+    { RC_LH_SOUTHWEST_GOSSIP_STONE, {RC_LH_SOUTHWEST_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_LAKE_HYLIA, "Southwest Gossip Stone", "LH Southwest Gossip Stone"}},
+    { RC_LW_GOSSIP_STONE, {RC_LW_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_LOST_WOODS, "Gossip Stone", "LW Gossip Stone"}},
+    { RC_LW_NEAR_SHORTCUTS_GOSSIP_STONE, {RC_LW_NEAR_SHORTCUTS_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_LOST_WOODS, "Near Shortcuts Gossip Stone", "LW Near Shortcuts Gossip Stone"}},
+    { RC_SFM_MAZE_LOWER_GOSSIP_STONE, {RC_SFM_MAZE_LOWER_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_SACRED_FOREST_MEADOW, "Maze Lower Gossip Stone", "SFM Maze Lower Gossip Stone"}},
+    { RC_SFM_MAZE_UPPER_GOSSIP_STONE, {RC_SFM_MAZE_UPPER_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_SACRED_FOREST_MEADOW, "Maze Upper Gossip Stone", "SFM Maze Upper Gossip Stone"}},
+    { RC_SFM_SARIA_GOSSIP_STONE, {RC_SFM_SARIA_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_SACRED_FOREST_MEADOW, "Saria Gossip Stone", "SFM Saria Gossip Stone"}},
+    { RC_TOT_LEFT_CENTER_GOSSIP_STONE, {RC_TOT_LEFT_CENTER_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_MARKET, "ToT Left Center Gossip Stone", "ToT Left Center Gossip Stone"}},
+    { RC_TOT_LEFT_GOSSIP_STONE, {RC_TOT_LEFT_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_MARKET, "ToT Left Gossip Stone", "ToT Left Gossip Stone"}},
+    { RC_TOT_RIGHT_CENTER_GOSSIP_STONE, {RC_TOT_RIGHT_CENTER_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_MARKET, "ToT Right Center Gossip Stone", "ToT Right Center Gossip Stone"}},
+    { RC_TOT_RIGHT_GOSSIP_STONE, {RC_TOT_RIGHT_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_MARKET, "ToT Right Gossip Stone", "ToT Right Gossip Stone"}},
+    { RC_ZD_GOSSIP_STONE, {RC_ZD_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_DOMAIN, "Gossip Stone", "ZD Gossip Stone"}},
+    { RC_ZR_NEAR_DOMAIN_GOSSIP_STONE, {RC_ZR_NEAR_DOMAIN_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Near Domain Gossip Stone", "ZR Near Domain Gossip Stone"}},
+    { RC_ZR_NEAR_GROTTOS_GOSSIP_STONE, {RC_ZR_NEAR_GROTTOS_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Near Grottos Gossip Stone", "ZR Near Grottos Gossip Stone"}},
+    { RC_ZR_OPEN_GROTTO_GOSSIP_STONE, {RC_ZR_OPEN_GROTTO_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Open Grotto Gossip Stone", "ZR Open Grotto Gossip Stone"}},
+};
+
+std::map<RandomizerCheckArea, std::string> rcAreas = {
```

Comment:
I'm guessing this is a `map` instead of an `unordered_map` because we want the areas to be in a specific order in the tree? In that case, should `RandomizerCheckObjects` also be a `map`? Or is there already some other logic in place I'm just not seeing to sort them alphabetically?

Reply:
https://github.com/HarbourMasters/Shipwright/pull/1249/commits/dd9acbbb43e56905dfd07b9ee962e7307b825e52